pkg/k8s/defaulter.go

```diff
@@ -45,28 +45,54 @@ func NewStateIntoSpecDefaulter(client client.Client) Defaulter {
 func (v *StateIntoSpecDefaulter) ApplyDefaults(ctx context.Context, resource client.Object) (changed bool, err error) {
```

Comment:
As we build this out for both CC & CCC, I'm going to steal Justin's feedback point and say that it may read easier to fetch both CC & CCC in advance and then decide what to how to default. We could use the [`resourceactuation.FetchLiveState()`](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/resourceactuation/resourceactuation.go#L35-L65) . So something like:

```
func (v *StateIntoSpecDefaulter) ApplyDefaults(ctx context.Context, resource client.Object) (changed bool, err error) (changed bool, err error) {
  ...
  cc, ccc, err := resourceactuation.FetchLiveKCCState()
  // defaulting based on those cc, ccc
  ...
}
```

PS: It may be that `pkg/controller/resourceactuation` is not the best home for `FetchLiveKCCState` but rather `pkg/k8s` .

---

Additionally, I also think we have an opportunity to add lowed down the pyramid tests here: `defaulter_test.go`. Then we can check for a placeholder resource that the precedence between CC & CCC is right when applying the default.

Reply:
> PS: It may be that pkg/controller/resourceactuation is not the best home for FetchLiveKCCState but rather pkg/k8s .

Actually I moved it to operator/pkg/livestate to avoid circular dependency issue considering the fetch is for operator types only. @acpana wdyt?